scripts/cnn_variant_wdl/vcf_analysis.R

```diff
@@ -0,0 +1,218 @@
+num_bins <- 50
+bin_by_quantile <-  FALSE
+
+get_proportion <- function(d, num_bins, column_to_sum, quality_column) {
+  x <- rowsum(column_to_sum, quality_column, na.rm =T)
+  idx <- row.names(x)
+  
+  for (i in 1:num_bins) {
+    qsum <- sum(quality_column==as.numeric(idx[i]))
+    if (!is.na(x[i]) && qsum>0) {
+      x[i] <- x[i] / qsum
+    }
+  }
+  return(x[quality_column])
+}
+
+print("try to merge.")
+d <- merge(d, dt, by=c("CHROM", "POS", "REF", "ALT"))
+d$TP <- as.numeric(d$CONC_ST!="FP,TN" & d$CONC_ST!="FP" & d$CONC_ST!="EMPTY")
+d$True_Positive <- d$CONC_ST!="FP,TN" & d$CONC_ST!="FP" & d$CONC_ST!="EMPTY"
+d$Unfiltered <- d$FILTER == "PASS" | d$FILTER == "."
+d$SNP <- d$EVENTLENGTH == 0
+d$ONE <- 1
+x <- rowsum(d$ONE, d$EVENTLENGTH)
+d$EVENTLENGTH_SUM <- x[as.factor(d$EVENTLENGTH)]
+d$Variant_Type <- paste(d$TYPE, as.factor(d$EVENTLENGTH<0))
+d$Truth_Status <- ifelse(d$True_Positive & d$Unfiltered, "True Positive", ifelse(d$True_Positive & !d$Unfiltered, "False Negative", ifelse(!d$True_Positive & d$Unfiltered, "False Positive", "True Negative")))
+statusColor <- c("True Positive" = "springgreen3", "True Negative" = "aquamarine4", "False Positive" = "red", "False Negative" = "orange")
+
+# All variant plots
+print("Make all variant plots.")
+p1 <- ggplot(d, aes(get(score_key), color=SNP, fill=SNP)) + 
+  scale_fill_discrete(name="Variant\nType", breaks=c("TRUE", "FALSE"), labels=c("SNPs", "INDELs")) +
+  geom_density(alpha=0.55) +
+  ggtitle(plot_title) +
+  xlab(score_label) +
+  guides(color=FALSE)
+
+p2 <- ggplot(d, aes(x=EVENTLENGTH, y=get(score_key), group=EVENTLENGTH, color=Truth_Status, shape=Variant_Type)) + 
+  scale_color_manual(values=statusColor) +
+  scale_shape_discrete(name='', breaks=c("INDEL TRUE", "INDEL FALSE", "SNP FALSE"), labels=c("Deletion", "Insertion", "SNP")) +
+  geom_jitter(height = 0, width = 0.1, alpha=0.6) + 
+  ggtitle(plot_title) +
+  ylab(score_label) + 
+  xlab("Event Length: - Deletions, 0 SNPs, + Insertions")
+
+p3 <- ggplot(d, aes(x=EVENTLENGTH, y=get(score_key), group=EVENTLENGTH, color=Truth_Status)) + xlim(-20, 20) + 
+  scale_color_manual(values=statusColor) +
+  geom_jitter(height = 0, width = 0.1, alpha=0.4) + 
+  geom_violin(color="grey", alpha=0) + 
+  geom_text(aes(x=EVENTLENGTH, y=14, label=EVENTLENGTH_SUM), color="grey30", size=2, angle=60) +
+  ggtitle(plot_title) +
+  ylab(score_label) + 
+  xlab("Event Length: - Deletions, 0 SNPs, + Insertions")
+
+p4 <- ggplot(d, aes(x=EVENTLENGTH, y=get(score_key), group=EVENTLENGTH, color=Truth_Status)) + xlim(-10, 10) + 
+  scale_color_manual(values=statusColor) +
+  geom_jitter(height = 0, width = 0.2, alpha=0.4) + 
+  geom_violin(color="grey", alpha=0) + 
+  geom_text(aes(x=EVENTLENGTH, y=14, label=EVENTLENGTH_SUM), color="grey30", size=3, angle=30) +
+  ylab(score_label) + 
+  xlab("Event Length: - Deletions, 0 SNPs, + Insertions")
+
+p5 <- ggplot(d, aes(x=EVENTLENGTH, y=get(score_key), group=EVENTLENGTH, color=Truth_Status)) + 
```

Comment:
This is insertions and deletions lumped together?  But won't deletions still have negative event length?  How is this different from p2?

Reply:
This is limited to INDELs <= 5BP long